src/libs/shared/src/common/domain/entities/artifact_ingestion.rs

```diff
@@ -120,6 +120,9 @@ impl ArtifactIngestion {
             return Err(IngestionError::ArtifactPath("The artifact_path must be set before moving the ingestion into a Finished state".into()));
         }
 
+        // Changes the status
+        self.status = new_status;
```

Comment:
In a function that changes the status, I missed actually changing the status. I love it